ml-agents-envs/mlagents_envs/base_env.py

```diff
@@ -247,90 +246,58 @@ def empty(spec: "BehaviorSpec") -> "TerminalSteps":
 class ActionType(Enum):
     DISCRETE = 0
     CONTINUOUS = 1
+    HYBRID = 2
 
 
-class BehaviorSpec(NamedTuple):
-    """
-    A NamedTuple to containing information about the observations and actions
-    spaces for a group of Agents under the same behavior.
-     - observation_shapes is a List of Tuples of int : Each Tuple corresponds
-     to an observation's dimensions. The shape tuples have the same ordering as
-     the ordering of the DecisionSteps and TerminalSteps.
-     - action_type is the type of data of the action. it can be discrete or
-     continuous. If discrete, the action tensors are expected to be int32. If
-     continuous, the actions are expected to be float32.
-     - action_shape is:
-       - An int in continuous action space corresponding to the number of
-     floats that constitute the action.
-       - A Tuple of int in discrete action space where each int corresponds to
-       the number of discrete actions available to the agent.
-    """
-
-    observation_shapes: List[Tuple]
-    action_type: ActionType
-    action_shape: Union[int, Tuple[int, ...]]
+class ActionSpec(NamedTuple):
+    num_continuous_actions: int
+    discrete_branch_sizes: Tuple[int, ...]
 
+    # For backwards compatibility
     def is_action_discrete(self) -> bool:
         """
         Returns true if this Behavior uses discrete actions
         """
-        return self.action_type == ActionType.DISCRETE
+        return self.discrete_action_size > 0
 
+    # For backwards compatibility
     def is_action_continuous(self) -> bool:
         """
         Returns true if this Behavior uses continuous actions
         """
-        return self.action_type == ActionType.CONTINUOUS
+        return self.continuous_action_size > 0
+
+    @property
+    def discrete_action_branches(self) -> Tuple[int, ...]:
```

Comment:
I agree with Vince's point further down that these names shouldn't have the word `action` in them since they're part of an `action_spec` (and it's implied). So you'd have `behavior_spec.action_spec.discrete_branches` or `action_spc.discrete_size`.

Reply:
Good call